src/dual.jl

```diff
@@ -211,45 +231,41 @@ end
     return Dual(v1 / v2, _div_partials(partials(n1), partials(n2), v1, v2))
 end
 
-@inline function @compat(Base.:/)(x::Real, n::Dual)
+@ambiguous @inline function @compat(Base.:/)(x::Real, n::Dual)
     v = value(n)
     divv = x / v
     return Dual(divv, -(divv / v) * partials(n))
 end
 
-@inline @compat(Base.:/)(n::Dual, x::Real) = Dual(value(n) / x, partials(n) / x)
+@ambiguous @inline @compat(Base.:/)(n::Dual, x::Real) = Dual(value(n) / x, partials(n) / x)
 
 # Exponentiation #
 #----------------#
 
 for f in (macroexpand(:(@compat(Base.:^))), :(NaNMath.pow))
     @eval begin
         @ambiguous @inline function ($f){N}(n1::Dual{N}, n2::Dual{N})
-            if isconstant(n2)
-                return $(f)(n1, value(n2))
-            else
-                v1, v2 = value(n1), value(n2)
-                expv = ($f)(v1, v2)
-                powval = v2 * ($f)(v1, v2 - 1)
-                logval = expv * log(v1)
-                new_partials = _mul_partials(partials(n1), partials(n2), powval, logval)
-                return Dual(expv, new_partials)
-            end
+            v1, v2 = value(n1), value(n2)
+            expv = ($f)(v1, v2)
+            powval = v2 * ($f)(v1, v2 - 1)
+            logval = isconstant(n2) ? one(expv) : expv * log(v1)
```

Comment:
This `isconstant(n2)` branch is necessary for the following to work:

```
julia> a, b = Dual(-2.0,1.0), Dual(2.0,0.0)
(Dual(-2.0,1.0),Dual(2.0,0.0))

julia> a ^ b
Dual(4.0,-4.0)
```

This case usually turns up when upstream type promotion has converted some scalar `x::Real` to `Dual{N,typeof(x)}` (with all zero partial components).

My slight refactor here is because this PR somehow causes the compiler to give up on inferring the original branch for nested `Dual`s (AFAICT it's fully inferrable, Julia is just being wonky). Unfortunately,  in its refactored position, this branch now partially prevents SIMD vectorization of this method with nested `Dual` arguments, thereby causing the current test failure.

Does anybody know a good workaround here? If worst comes to worst, I'll accept the minor SIMD regression to enable this PR's functionality, but I'm hoping I can have my cake and eat it too.

@mlubin @KristofferC  curious if you guys have any insights